tests/integration/targets/backup_vault/tasks/main.yml

```diff
@@ -67,7 +70,23 @@
           - backup_vault_result_idem is not changed
           - backup_vault_result_idem.vault.backup_vault_name == backup_vault_name
           - backup_vault_result_idem.vault.encryption_key_arn == key.key_arn
- 
+          - backup_vault_result_idem.vault.tags.environment == "dev"
```

Comment:
We generally used to add several tests for tags:

- Add tags in check_mode
- Add tags
- Add the same tags (idempotency) in check_mode
- Add the same tags (idempotency)
- Update tags in check_mode
- Update tags 
- Update tags (idempotency) in check_mode
- Update tags (idempotency)
- Similar tests with purge_tags
For example https://github.com/ansible-collections/amazon.aws/blob/main/tests/integration/targets/ec2_vpc_igw/tasks/main.yml

Reply:
I'm happy to add more test coverage for tagging as part of this PR...however we are also missing both check and idempotency tests for all of the update and delete functionality in this module, and we don't have that level of tagging integration test coverage for any of the other recently added backup modules either. Would it be ok if we create a separate Jira ticket (or tickets) to improve the robustness of tests in these modules in general?

Reply:
@alinabuzachis I'm so glad you suggested this, adding tests found one more bug! I'll go ahead and create Jira tickets for improving tests on the other modules.